### ISSUE TITLE
why all my output results of the fcn32 is ZERO

### ISSUE THREAD

**Issue by daiyp3360xxxxxisxxxxxxmine** [wkentaro/pytorch-fcn#113]
after the fcn32 was trained for 9 epochs, the output value score of it is still zero. 
however, the weight of each layers are not zero. i am wondering if somebody has suggestions to fix this problem.

**Comment by wkentaro** [comment]
I think you should try using training data as validation data, and check if it works. If it doesn't work, that means you have a serious bug in your code.

**Comment by jdavidd** [comment]
I have the sane problem with fcn8s, help anyone?

**Comment by BCWang93** [comment]
> after the fcn32 was trained for 9 epochs, the output value score of it is still zero.
> however, the weight of each layers are not zero. i am wondering if somebody has suggestions to fix this problem.

Hi,Did you solved this problem?I trained 50 epoch,but when I use the trained model to test,the output is zero.

**Comment by BCWang93** [comment]
> I have the sane problem with fcn8s, help anyone?

Hi,Did you solved this problem?I trained 50 epoch,but when I use the trained model to test,the output is zero.

**Comment by wkentaro** [comment]
> I have the sane problem with fcn8s, help anyone?
 

For training fcn8s, you first start training fcn32s and fcn16s.

**Comment by BCWang93** [comment]
> > I have the sane problem with fcn8s, help anyone?
> 
> For training fcn8s, you first start training fcn32s and fcn16s.

Hi,I trained the fcn32 network,but I also has this problem,the output of the network is always zero.

**Comment by imxyu** [comment]
> > I have the sane problem with fcn8s, help anyone?
> 
> Hi,Did you solved this problem?I trained 50 epoch,but when I use the trained model to test,the output is zero.

I met the same problem, have you solved it?

**Comment by ShawnNew** [comment]
I found that the intermediate output of the network yields to zero after `conv1_1`, which uses a `padding=100`. Does anyone knows why this strange padding setting is used?

**Comment by ShawnNew** [comment]
I found that in the initialization function, weights are initialized to zero, which may cause the problem. Does anyone knows why the fcn8s need to initialize all the weights to zero?

**Comment by wkentaro** [comment]
> I found that in the initialization function, weights are initialized to zero, which may cause the problem. Does anyone knows why the fcn8s need to initialize all the weights to zero?

Because of this https://github.com/wkentaro/pytorch-fcn/issues/40

**Comment by daiyp3360** [comment]
I didn't solve this problem yet, for these days I was tracked by some trifles, I am just going to resume this work. I do think this problem might caused by the initial strategy. i will response here as soon as i solve this problem.

**Comment by BCWang93** [comment]
> I didn't solve this problem yet, for these days I was tracked by some trifles, I am just going to resume this work. I do think this problem might caused by the initial strategy. i will response here as soon as i solve this problem.

Hi，Did you solved this problem?Thanks!

**Comment by phucvu460** [comment]
I have the same problem with fcn32s, all output results are zeros. Anyone can help ?

**Comment by prfans** [comment]
I have the same problem, all output results are zeros. Anyone can help ?

**Comment by songyuc** [comment]
Hi, @daiyp3360xxxxxisxxxxxxmine , 
have figure out the problem yet?

**Comment by songyuc** [comment]
Hi @wkentaro ,
I think #40 may have no relation to the initiation strategy, so it cannot account for the problem here.

**Comment by hylovezhang** [comment]
I have same problem, did you solve it? @daiyp3360xxxxxisxxxxxxmine

**Comment by liul-ing** [comment]
can you run the code?i have lots of problems about this code.Could you please share me your code?My email：1562112553@qq.com。Thank you very much!!!
